### PR TITLE
Apply changes in foundry-primitives to use Ed25519

### PR DESCRIPTION
**Pull request by somniumism**

### CODE REVIEW

**Comment by somniumism** [comment]
Now I'm working on modifying the tests.

**Comment by somniumism** [comment]
@HoOngEe A total of 12 test failures occur. The test failures are as follows:
```
migrate › migrate

platform › importKey
platform › importKeyWithMeta

HD wallet test › get public key & private key - "m/0"
HD wallet test › get public key & private key - "M/0"
HD wallet test › get public key & private key - "m/0/1/2/3/4"
HD wallet test › get public key & private key - "m/0'"
HD wallet test › get public key & private key - "m/0'/1'/2'/3'/4'"
HD wallet test › import fixed raw seed and check: "d0a8182a44b71802ba85d35fad626c6275a82afb943dfee10d851140a61f74df"
HD wallet test › import fixed raw seed and check: "0000000000000000000000000000000000000000000000000000000000000000"
HD wallet test › import fixed raw seed and check: "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
HD wallet test › import fixed raw seed and check: "ffffffffffffffffffffffffffffffff"
```

File: src/__test__/hdwallet.spec.ts [hunk 1]
```diff
@@ -146,6 +147,7 @@ describe("HD wallet test", () => {
             "11fca78e1985e951c9cc7a866626741cefe972a24dd1b68a5c53940436b33a026e0c4736b6b6bc4b3b50600cf8c780e4ca01772272d3664939c571d29fa5a1d800"
         ]
     ])(
+        // FIXME
```
Comment (somniumism) [inline src/__test__/hdwallet.spec.ts]: I don't know how to make a test oracle like the example above.

File: src/__test__/platform.spec.ts [hunk 2]
```diff
@@ -14,16 +14,17 @@ describe("platform", () => {
...
             privateKey,
             passphrase: "satoshi"
         });
         const publicKey =
-            "0eb7cad828f1b48c97571ac5fde6add42a7f9285a204291cdc2a03007480dc70639d80c57d80ba6bb02fc2237fec1bb357e405e13b7fb8ed4f947fd8f4900abd";
+            "40200442f046f5012df6d6acf8105be68278a6d3bd631603ab86fed89cdf9901";
         expect(key).toBe(keyFromPublicKey(KeyType.Platform, publicKey));
     });
 
+    // FIXME
```
Comment (somniumism) [inline src/__test__/platform.spec.ts]: If there is meta data as shown below, how can I generate a public key for testing?

File: src/__test__/hdwallet.spec.ts (hunk 1 above)
Comment (somniumism) [reply src/__test__/hdwallet.spec.ts]: @HoOngEe Could you give me good advice? Can I create a testing oracle about this in the rust implementation?

File: src/__test__/platform.spec.ts (hunk 2 above)
Comment (seonpyoKim) [reply src/__test__/platform.spec.ts]: To generate the encrypted secret, you can utilize the subcommand `account`.  `./target/debug/foundry account --help` will helps

Comment (somniumism) [reply src/__test__/platform.spec.ts]: I'll check it : )

File: src/__test__/hdwallet.spec.ts (hunk 1 above)
Comment (seonpyoKim) [reply src/__test__/hdwallet.spec.ts]: It is much more simple in this case, you can expect each function's behavior by reading the code and generate oracles from python implementation of ed25519 or our primitives.

**Comment by somniumism** [comment]
In migrate.spec.ts test(migrate), `KeystoreError: DecryptionFailed` occurs. It is related to function `decode()` in storage.ts, and this line.
```
110        if (mac !== json.crypto.mac) {
111            throw new KeystoreError(ErrorCode.DecryptionFailed);
112        }
```

**Comment by HoOngEe** [comment]
Let's wait for the `primitives - 0.2.0` and make it to use that version.

**Comment by HoOngEe** [comment]
I think we can remove the dependency `bitcore-mnemonic`, check the usage please. Also, you should check the test fail.

**Review by HoOngEe — APPROVED** [review]

**Pull request merged**